### DISCUSSION TITLE
Solidity 101 Section 1: Simple Storage (Problems deploying to zksync)

### DISCUSSION THREAD

**Discussion by Heavens01** [Cyfrin/foundry-full-course-cu#2734] [Q&A]
This message "You are using an outdated version of the zkSync plugin: v0.6.1-7, please update to v0.6.2-24 by force-refreshing the page and clearing your browser cache." keeps popping up even after i've forced refreshed the page and cleared my browser cache. It isn't updating to the lastest version, What can i do please?

**Comment by EngrPips** [comment]
Hello, Have you tried uninstalling and re-installing the `zksync` plugin? Let's see if that will help.

**Reply by Heavens01** [reply]
Installing? I don't think so. I did deactivated the plugin several times all to no avail.

**Reply by EngrPips** [reply]
I think you can ignore that error and proceed with your work. I also got that error and I just cleared the terminal and proceeded with my work, and everything went well

**Reply by Kallyfive** [reply]
I'm facing the same issue, trying to ignore it but it is preventing me from deploying the contract. Here is the message

 "underlying network changed (event=\"changed\", network={\"chainId\":324,\"name\":\"zkSync Era\"}, detectedNetwork={\"name\":\"sepolia\",\"chainId\":11155111,\"ensAddress\":null}, code=NETWORK_ERROR, version=providers/5.7.2)"

**Reply by EngrPips** [reply]
Did you set your `metamask` wallet chain to `zksync-sepolia`? @Kallyfive

**Reply by Kallyfive** [reply]
Yes mate! @EngrPips

**Comment by Kallyfive** [comment]
Yes, I did

**Reply by EngrPips** [reply]
Then it is supposed to work as expected. Maybe you can show the full page of your remix workspace so I can have a look and see if I can spot anything weird.

**Reply by EngrPips** [reply]
> Then it is supposed to work as expected. Maybe you can show the full page of your remix workspace so I can have a look and see if I can spot anything weird.

This

**Reply by Kallyfive** [reply]
![Screenshot 2024-10-28 221252](https://github.com/user-attachments/assets/31057fd9-2e23-415d-ba51-1182e673ef69)

**Reply by EngrPips** [reply]
I am having a difficult time loading `Remix` but I will try this process as soon as I can and see how things go for me.

**Reply by engrusmanbelloa** [reply]
How far were you able to figure it out? @EngrPips

**Reply by EngrPips** [reply]
I haven't actually used Remix today, I will try and let you know as soon as possible.

**Comment by BowtiedHarpyEagle** [comment]
Also notice ZKSync stuff is still buggy and optional, so feel free to just move on with the course, even if it doesn't work.

**Reply by Kallyfive** [reply]
Thanks

**Reply by Heavens01** [reply]
Mine has just updated itself to the latest version and is operational now. I didn't do anything again apart from what I said i did tho.

**Reply by EngrPips** [reply]
Awesome.

**Comment by Heavens01** [comment]
@EngrPips Something else just came up please. I ticked the "verify contract" option and deployed the contract and popped out this message below in the terminal:

Error in plugin @matterlabs/hardhat-zksync-verify: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
ZkSyncVerifyPluginError: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
    at inferContractArtifacts (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/plugin.ts:48:45)
    at async OverriddenTaskDefinition.getContractInfo [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:310:31)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verifyContract [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:197:54)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verify [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:83:5)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
>

WHAT SHOULD I DO PLEASE??

**Reply by Heavens01** [reply]
What I'm trying to say in summary is I can't verify my contract.

**Reply by EngrPips** [reply]
Hello @Heavens01, do you have your solidity files in a `contract` folder?

**Reply by Heavens01** [reply]
Sure, but I've moved past that course. My problem is that I can deploy but can't verify my contract with the zkSync plugin.

**Comment by Heavens01** [comment]
@EngrPips Another quick problem please, I've been unable to enter the course since yesternight. I mean this url "https://updraft.cyfrin.io/".
Is it undergoing an upgrade or is there something else please.

**Reply by EngrPips** [reply]
Hello @Heavens01, Are you still facing this problem?

**Reply by Heavens01** [reply]
Not really. The site is denying a certain network SIM (Glo SIM Network) access to serve as a connection source which i happen to be using. I'll have to use another SIM that isn't glo but they should fix that.

**Reply by EngrPips** [reply]
Oh I see.

**Reply by Heavens01** [reply]
@EngrPips Hey man, been a while. I'm having a hard time figuring out what to do here "Solidity 101 Section 1 NFTs". i don't understand how to interact with it to get the nft.

**Reply by EngrPips** [reply]
Read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2018 , It should help you understand how to solve the challenges

**Comment by Heavens01** [comment]
Good morning @EngrPips or anyone, and happy Christmas.

**Reply by EngrPips** [reply]
Good morning, Friend. How are you doing?

**Reply by Heavens01** [reply]
Wow, I'm fine. Wasn't hoping you'd reply this soon.
Smiles

**Reply by EngrPips** [reply]
I am always trying to attended to things that needs attention.

**Reply by Heavens01** [reply]
Thanks for your help so far. @EngrPips

**Reply by Heavens01** [reply]
I want to follow you on X and I'm also new to using X so how do we go about that?

**Reply by EngrPips** [reply]
You are always welcome, Friend.  Are you on Cyfrin Discord? I am more active on Discord than on twitter.

**Comment by Heavens01** [comment]
![image](https://github.com/user-attachments/assets/5c3d551b-25fd-4482-86db-96e9e0f7462c)
@EngrPips Please which Distro should i use exactly? I just entered Foundry newly.

**Reply by EngrPips** [reply]
Hello @Heavens01, I never used `wsl`, but I will suggest you download `Ubuntu 24.02 LTS`
